File: sem3/p3v2/calculate_volume.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
int count_hits(float r, int *hits, int N);
float randomFloat(float min, float max);
int calculate_volume(float *volume, float r, int N, int seed);

int main() {
    return 0;
}
float randomFloat(float min, float max){
    return (float)(rand()) / (float)(RAND_MAX) * (max - min) + min;
}
int calculate_volume(float *volume, float r, int N, int seed){
    srand(seed);
    float cube_volume = pow((2*r), 3);
    int hits = 0;
    count_hits(r, &hits, N);
    *volume = cube_volume * (float) hits/N;
    return 0;
}
int count_hits(float r, int *hits, int N){
    for(int i = 0; i < N; i++){
        float x = randomFloat(-r, r);
        float y = randomFloat(-r, r);
        float z = randomFloat(-r, r);
        if (((x*x + y*y) <= r*r) && ((x*x + z*z) <= r*r))
            (*hits)++;
    }
    return 0;
}

File: sem3/p3v2/calculate_volume.py
import numpy as np

def calculate_volumne(r: float, N: int, seed: int) -> float:
    np.random.seed(seed)
    cube_volume = (2*r)**3
    volume = cube_volume * count_hits(r, N)/N
    return volume

def count_hits(r:float, N: int) -> int:
    hits = 0
    for i in range(1,N):
        x = np.random.uniform(-r,r)
        y = np.random.uniform(-r,r)
        z = np.random.uniform(-r,r)
        if (x**2 + y**2 <= r*r and x**2 + z**2 <= r*r): hits +=1
    return hits
